Swift 4.2

maurovc opened the pull request.

Commit:
Swift 4.2

harlanhaskins commented on Pod/Classes/Banner.swift:
```diff
@@ -114,18 +128,32 @@ open class Banner: UIView {
-        label.font = UIFont.preferredFont(forTextStyle: UIFontTextStyle.headline)
+      
+        #if swift(>=4.2)
+        let textStyle = UIFont.TextStyle.headline
```
This one can just be changed to ` UIFont.preferredFont(forTextStyle: .headline)` and be backwards-compatible.

maurovc replied:
True

harlanhaskins commented on Pod/Classes/Banner.swift:
```diff
@@ -114,18 +128,32 @@ open class Banner: UIView {
-    @objc open let detailLabel: UILabel = {
+    @objc public let detailLabel: UILabel = {
         let label = UILabel()
-        label.font = UIFont.preferredFont(forTextStyle: UIFontTextStyle.subheadline)
```
Ditto here.

maurovc replied:
+1

harlanhaskins commented on Pod/Classes/Banner.swift:
```diff
@@ -277,7 +305,14 @@ open class Banner: UIView {
+            let options = NSLayoutFormatOptions()
+            #endif
+          
+            contentView.addConstraints(NSLayoutConstraint.defaultConstraintsWithVisualFormat(constraintFormat, options: options, metrics: nil, views: ["label": view]))
```
I think you can change this to `[]` and still be backwards compatible.

harlanhaskins commented on Pod/Classes/Banner.swift:
```diff
@@ -369,11 +404,25 @@ open class Banner: UIView {
 
 extension NSLayoutConstraint {
+  #if swift(>=4.2)
+  @objc class func defaultConstraintsWithVisualFormat(_ format: String, options: NSLayoutConstraint.FormatOptions = NSLayoutConstraint.FormatOptions(), metrics: [String: AnyObject]? = nil, views: [String: AnyObject] = [:]) -> [NSLayoutConstraint] {
```
Same here

maurovc replied:
+1

harlanhaskins commented:
Thank you for this!

Commit:
Cleanup and back to 4.0

Commit:
More cleanup

maurovc commented:
Hey! @harlanhaskins I've updated the PR with some of your comments and moved the project again to Swift 4.0 so you're able to merge the code to the main branch.

nmanoogian commented:
Thank you for your contribution, @maurovc! Several people submitted PRs for this migration and they were all very similar. Version 0.8.2 is now available with these changes.
